.github/workflows/detekt-actions.yml

```diff
@@ -0,0 +1,27 @@
+name: Build-actions
+
+on:
+  push:
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'oracle'
+          java-version: '17'
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Compile and run checks
+        run: ./gradlew build
```

Comment:
let's rename the file. `detekt-actions` doesn't trigger only detekt. 

`build` gradle task runs all checks